app/lib/features/cross_signing/cross_signing.dart

```diff
@@ -1,874 +1,468 @@
 import 'dart:async';
-import 'dart:io';
 
-import 'package:acter/common/themes/app_theme.dart';
-import 'package:acter/router/router.dart';
+import 'package:acter/features/cross_signing/pages/request_cancelled.dart';
+import 'package:acter/features/cross_signing/pages/request_created.dart';
+import 'package:acter/features/cross_signing/pages/request_done.dart';
+import 'package:acter/features/cross_signing/pages/request_ready.dart';
+import 'package:acter/features/cross_signing/pages/sas_accepted.dart';
+import 'package:acter/features/cross_signing/pages/sas_cancelled.dart';
+import 'package:acter/features/cross_signing/pages/sas_keys_exchanged.dart';
+import 'package:acter/features/cross_signing/pages/sas_started.dart';
+import 'package:acter/features/cross_signing/pages/verification_request.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
-import 'package:atlas_icons/atlas_icons.dart';
+import 'package:events_emitter/events_emitter.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_gen/gen_l10n/l10n.dart';
-import 'package:go_router/go_router.dart';
-import 'package:sprintf/sprintf.dart';
-
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:logging/logging.dart';
 
 final _log = Logger('a3::cross_signing');
 
-class VerificationProcess {
-  bool verifiyingThisDevice;
-  String stage;
-
-  VerificationProcess({
-    required this.verifiyingThisDevice,
-    required this.stage,
-  });
-}
-
 class CrossSigning {
-  Client client;
-  bool acceptingRequest = false;
-  bool waitForMatch = false;
-  late StreamSubscription<VerificationEvent>? _verificationPoller;
-  final Map<String, VerificationProcess> _processMap = {};
-  bool _mounted = true;
-  bool isDesktop = Platform.isWindows || Platform.isMacOS || Platform.isLinux;
-
-  CrossSigning({required this.client}) {
-    _installVerificationEvent();
-  }
+  final Client client;
+  final Ref ref;
 
-  void dispose() {
-    _mounted = false;
-    _verificationPoller?.cancel();
-  }
+  final EventEmitter _verifEmitter = EventEmitter();
+  bool _isVerifier = false; // whether this device requested verification
+  String? _flowId; // if not null, verification flying now
+  bool _keyExchanged = false;
+  Stream<VerificationEvent>? _verifListener;
+  StreamSubscription<VerificationEvent>? _verifPoller;
 
-  void _installVerificationEvent() {
-    _verificationPoller = client.verificationEventRx()?.listen((event) {
+  CrossSigning(this.client, this.ref) {
+    _verifListener = client.verificationEventRx(); // keep it resident in memory
+    _verifPoller = _verifListener?.listen((event) {
```

Comment:
I don't quite understand the point of this `CrossSigning` object and it's `EventEmitter`. Why isn't this just wrapped into a [riverpod `StreamProvider`](https://riverpod.dev/docs/providers/stream_provider)? Seems to much easier, less code, better integrated into our actual system and doesn't need the newly added event_emitter dependency...

Reply:
Agree, could use ```StreamProvider``` for verification stream here with having ref listener and do appropriate calls based on that...

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1257/commits/c9c0569cca442daf8549b67deae5e9980bf7d884